File: Lua/pe003.lua
--[[
PE003.lua
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
--]]


--[[
function ppfp(n) --renvoie le plus petit facteur premier
  for i=2,n^(0.5),1 do
    if n%i==0 then return i end
  end
  return n
end
--]]

dofile("DIVlib.lua")

nombre=600851475143
--print("pgfp=",pgfp(nombre))

print("Decomposition de " .. nombre .. ":")

local t={}

t=decompPrime(nombre,t)
print("-->" .. nombre .. " = " .. table.concat(t," ; "))

File: Python/PE003.py
##PE003
##
##The prime factors of 13195 are 5, 7, 13 and 29.
##
##What is the largest prime factor of the number 600851475143 ?


nb = 600851475143


def isPrime(n):
    if n <= 1:
        return False
    i = 2
    while i <= n ** 0.5:
        if n % i == 0:
            return False
        i+=1
    return True

def nextPrime(n):
    i = n + 1
    while not isPrime(i):
        i+=1
    return i

def primeFactors(n):
    l = []
    i = 2
    a = n
    while a != 1 and i <= a:
        if a % i == 0:
            l.append(i)
            a /= i
        else:
            i = nextPrime(i)
    return l

print(primeFactors(nb)[::-1])


    
  
  

